doc/ping.xml

```diff
@@ -107,17 +107,34 @@ xml:id="man.ping">
     arbitrary number of “pad” bytes used to fill out the
     packet.</para>
     <para>
-    <command>ping</command> works with both IPv4 and IPv6. Using
-    only one of them explicitly can be enforced by specifying
-    <option>-4</option> or
-    <option>-6</option>.</para>
-    <para>
     <command>ping</command> can also send IPv6 Node Information
     Queries (RFC4620). Intermediate
     <emphasis remap="I">hop</emphasis>s may not be allowed, because
     IPv6 source routing was deprecated (RFC5095).</para>
   </refsection>
 
+  <refsection xml:id="destination">
+    <info>
+      <title>DESTINATION</title>
```

Comment:
TBH I feel the content is a bit awkward but I'm affraid I'm just bike shedding. You've been working on this for a long time so I'll just let the maintainers chime in.

Reply:
On a positive note, I agree that a separate section is worthwhile to discuss interface specification with ipv6 link local destinations.

Reply:
I agree that the content of man page section I added in commit "man/ping: Move destination to own section, improve" is a bit awkward. I wanted to be clear about unfixed kernels and old iputils, but I don't like wording myself. Maybe just mentioned that it was fixed in 5.17 and in LTS kernels >= 4.14.

Reply:
Ok, here were my thoughts:

> Newer kernels (>= 5.17 and LTS: 5.15, 5.10, 5.4, 4.19, 4.14) contain

There will come a time in the future when those are no longer "newer kernels". It would also be more useful to mention the sublevel of the stable kernels, ie.: 5.15.19, 5.10.96, 5.4.176, 4.19.228, 4.14.265

> It also requires ping >= s20180629, therefore using

Generally, the man page is only relevant for the version of ping it is shipped with, right? I would skip that part.

How about:
```
IPV6 LINK-LOCAL DESTINATIONS

For IPv6, when the destination address has link-local scope and ping is using
ICMP datagram sockets, the output interface must be specified. When ping is
using raw sockets, it is not strictly necessary to specify the output
interface but it should be done to avoid ambiguity when there are multiple
possible output interfaces.

There are two ways to specify the output interface:
* using the % notation
	The destination address is postfixed with % and the output interface
	name or ifindex, for example:
		ping fe80::5054:ff:fe70:67bc%eth0
		ping fe80::5054:ff:fe70:67bc%2
* using the -I option
	When using ICMP datagram sockets, this method is only supported on
	some kernel versions (...).
```
The remaining issue IMO is that it is not easy for a user to know if ping is using ICMP sockets or raw sockets.

Reply:
Thanks, I'll use this, add you as author. I'd keep info in git commit message (also helps me to remember on changelog).

One more note to changelog: musl just does not support `-I iface` even with fixed kernel (I suppose `getaddrinfo()` is different). But OTOH it allows to ping even without `'%'-notation` (but works with the notation).

Reply:
> The remaining issue IMO is that it is not easy for a user to know if ping is using ICMP sockets or raw sockets.

I've been thinking about print this when `-v` or even always.
https://github.com/iputils/iputils/issues/411

Reply:
I added note about unsupported musl and test also uclibc-ng.

Reply:
OK, it works also on uclibc-ng. Thus only musl not supported (ignoring bionic although aosp thinks about updating iputils.